src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,114 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 32;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private int size;
+    private Entry<K, V>[] table;
+
+    public MyHashMap() {
+        this.table = new Entry[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeeded();
+        int hash = getIndex(key);
```

Comment:
```suggestion
        int index = getIndex(key);
```
![image](https://github.com/mate-academy/jv-hash-map/assets/20326858/59ee3da1-b94c-42d8-8cee-e05369902e36)